### PR TITLE
Refactor Ops

### PR DESCRIPTION
**Pull request by slaviered**
* Separate command executor from Ops interfce to allow mocking in unit tests
   * This is Part 3 of [MGMT-2403](https://issues.redhat.com/browse/MGMT-2403)

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -209,13 +209,7 @@ func (i *installer) extractIgnitionToFS(ignitionPath string) (err error) {
...
 	i.log.Infof("Extracting ignition to disk using %s mcoImage", mcoImage)
 	for j := 0; j < extractRetryCount; j++ {
-		_, err = i.ops.ExecPrivilegeCommand(utils.NewLogWriter(i.log), "podman", "run", "--net", "host",
-			"--volume", "/:/rootfs:rw",
-			"--volume", "/usr/bin/rpm-ostree:/usr/bin/rpm-ostree",
-			"--privileged",
-			"--entrypoint", "/usr/bin/machine-config-daemon",
-			mcoImage,
-			"start", "--node-name", "localhost", "--root-mount", "/rootfs", "--once-from", ignitionPath, "--skip-reboot")
+		err = i.ops.ExtractIgnitionWithMCO(mcoImage, ignitionPath)
```
Comment (carbonin) [inline src/installer/installer.go:212]: Maybe `ExtractIgnitionFromMCO`?

File: src/ops/ops.go [hunk 2]
```diff
@@ -56,6 +54,7 @@ type ops struct {
 	log       *logrus.Logger
 	logWriter *utils.LogWriter
 	cmdEnv    []string
+	Runner    opsexec.Excecutor
```
Comment (carbonin) [inline src/ops/ops.go:57]: Any reason not to use a pointer here?

File: src/ops/ops.go [hunk 3]
```diff
@@ -3,33 +3,31 @@ package ops
...
 	"strings"
-	"syscall"
 	"text/template"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/assisted-installer/src/config"
 	"github.com/openshift/assisted-installer/src/inventory_client"
+	opsexec "github.com/openshift/assisted-installer/src/ops/exec"
```
Comment (carbonin) [inline src/ops/ops.go:18]: Maybe `ops_exec` here?

File: src/ops/ops.go (hunk 2 above)
Comment (carbonin) [reply src/ops/ops.go:57]: If the goal is to "hide" the details of how we are actually running these commands, can this not be exported?
And maybe we use either "runner" or "executor" for both the type and the struct field to reduce the number of names we use for the same thing.

**Review by carbonin — COMMENTED** [review]
Just a few comments about naming.
Maybe also add a test that would not have been possible (or very difficult) to write before this change? Looks like `GetVGByPV` would be a good candidate for some tests.

**Comment by slaviered** [comment]
closing it for now. will not merge it with the main feature

**Pull request closed without merging**